File: Concentus.Celt/Structs/PulseCache.cs
using Concentus.Common.CPlusPlus;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Concentus.Celt.Structs
{
    public class PulseCache
    {
        public int size = 0;
        public Pointer<short> index = null;
        public Pointer<byte> bits = null;
        public Pointer<byte> caps = null;

        public void Reset()
        {
            size = 0;
            index = null;
            bits = null;
            caps = null;
        }
    }
}

File: go/opus/PulseCache.go
package opus

type PulseCache struct {
    size  int
    index []int16
    bits  []int16
    caps  []int16
}

func (p *PulseCache) Reset() {
    p.size = 0
    p.index = nil
    p.bits = nil
    p.caps = nil
}